syne_tune/backend/trial_backend.py

```diff
@@ -157,6 +179,8 @@ def fetch_status_results(self, trial_ids: List[int]) -> Tuple[Dict[int, Tuple[Tr
                     position_last_seen = self._last_metric_seen_index[trial_result.trial_id]
                     new_metrics = trial_result.metrics[position_last_seen:]
                     self._last_metric_seen_index[trial_result.trial_id] += len(new_metrics)
+                    if self.delete_checkpoints and trial_result.status == Status.completed:
```

Comment:
This code is not needed since you call the deletion already in l137.

Reply:
Hi David, the one in l137 is only triggered when the trial is stopped, not when it ends on its own. For example, with FIFOScheduler (RS, BO), trials are never stopped.

One important point to note: Calling delete_checkpoint for a trial which does not have a checkpoint, is not doing anything. So you can just call it multiple times for the same trial.

Reply:
Ok this makes sense. 

Can we deal with this case only at this place then? It seems that we also probably want to delete checkpoints in other termination cases (when the status becomes failed, stopped in addition to completed).